src/gallery.cpp

```diff
@@ -12,38 +12,59 @@ Gallery::Gallery(){
 //---- find or create gallery directory----
     pathname = getenv("HOME");
     pathname += + "/cellUviewGallery/"; 
+    flatFieldPath = pathname + ".FlatFieldGallery";
+    //updates index to find highest existing file with matching name to avoid overwriting
+    initialiseDirectory(pathname, "cellUview Gallery");
+    initialiseDirectory(flatFieldPath, "Flat field capture gallery");
+    updateIndex();
     
-    //if it doesn't exist
-    if ((dir = opendir(pathname.c_str())) == NULL){
+}
+
+int Gallery::initialiseDirectory(std::string path, std::string description){
+        //if it doesn't exist
+    if ((dir = opendir(path.c_str())) == NULL){
         //try to make the directory
-        if (mkdir(pathname.c_str(), S_IRWXU) == -1){
+        if (mkdir(path.c_str(), S_IRWXU) == -1){
             //if failed:
             std::cerr << "Error :  " << std::strerror(errno) << std::endl;
-            std::cout << "Gallery directory not found/created";
+            std::cout << description << " directory not found/created";
             //ADD. disable button if failed
             pathname = ""; 
-            return;
+            return 1;
         }
         else{//if gallery succesfully made
-            std::cout << "Gallery directory created at " + pathname << std::endl;
+            std::cout << description << " directory created at " + path << std::endl;
         }
     }else{//if gallery already exists
-        std::cout << "Gallery directory found at " + pathname << std::endl;
+        std::cout << description << " directory found at " + path << std::endl;
         closedir(dir);
     }
 
+    return 0;
+
     //updates index to find highest existing file with matching name to avoid overwriting
-    updateIndex();
-    
 }
 
+
 //add some error handling
-void Gallery::captureFrame(frame capFrame){
+void Gallery::captureFrame(frame capFrame, bool updateFlatField){
     if (pathname == ""){
         return;
     }
     //add ability to set custom string before number
-
+    if (updateFlatField == true){
+        for (int i = 1; i <= 20; i++) {
+            captureFname = pathname + "/.FlatFieldGallery/flatField" + std::to_string(i) + ".jpg";
```

Comment:
This doesn't do what you think it does, it captures the same frame 20 times, rather than 20 different frames.